packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/PreviewHostApiImpl.java

```diff
@@ -0,0 +1,142 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camerax;
+
+import android.graphics.SurfaceTexture;
+import android.util.Size;
+import android.view.Surface;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import androidx.camera.core.Preview;
+import androidx.camera.core.SurfaceRequest;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugins.camerax.GeneratedCameraXLibrary.PreviewHostApi;
+import io.flutter.view.TextureRegistry;
+import java.util.concurrent.Executors;
+
+public class PreviewHostApiImpl implements PreviewHostApi {
+  private final BinaryMessenger binaryMessenger;
+  private final InstanceManager instanceManager;
+  private final TextureRegistry textureRegistry;
+
+  @VisibleForTesting public CameraXProxy cameraXProxy = new CameraXProxy();
+  @VisibleForTesting public TextureRegistry.SurfaceTextureEntry flutterSurfaceTexture;
+
+  public PreviewHostApiImpl(
+      BinaryMessenger binaryMessenger,
+      InstanceManager instanceManager,
+      TextureRegistry textureRegistry) {
+    this.binaryMessenger = binaryMessenger;
+    this.instanceManager = instanceManager;
+    this.textureRegistry = textureRegistry;
+  }
+
+  /** Creates a {@link Preview} with the target rotation and resolution if specified. */
+  @Override
+  public void create(
+      @NonNull Long identifier,
+      @Nullable Long rotation,
+      @Nullable GeneratedCameraXLibrary.ResolutionInfo targetResolution) {
+    Preview.Builder previewBuilder = cameraXProxy.createPreviewBuilder();
+    if (rotation != null) {
+      previewBuilder.setTargetRotation(rotation.intValue());
+    }
+    if (targetResolution != null) {
+      previewBuilder.setTargetResolution(
+          new Size(
+              targetResolution.getWidth().intValue(), targetResolution.getHeight().intValue()));
+    }
+    Preview preview = previewBuilder.build();
+    instanceManager.addDartCreatedInstance(preview, identifier);
+  }
+
+  /**
+   * Sets the {@link Preview.SurfaceProvider} that will be used to provide a {@code Surface} backed
+   * by a Flutter {@link TextureRegistry.SurfaceTextureEntry} used to build the {@link Preview}.
+   */
+  @Override
+  public Long setSurfaceProvider(@NonNull Long identifier) {
```

Comment:
Consider giving some guidance on the expected formatting or source of `identifier`. 
Is it random? do the bits mean something? Is it expected to come from another system?

Reply:
ok, is there a class or doc that this can defer to for documentation? Even if it isn't public comments help people new to the team or those unfamiliar with plugins understand what is happening in code. 

This comment is non blocking.

Reply:
I agree this could be confusing. I think a better solution would be to include a section in the README overviewing how the approach is used in the plugin, since identifiers and other artifacts of the approach are used throughout. Filed an issue for this: https://github.com/flutter/flutter/issues/120103